common/src/main/java/com/sequenceiq/cloudbreak/logger/format/CustomJsonLayout.java

```diff
@@ -1,29 +1,65 @@
 package com.sequenceiq.cloudbreak.logger.format;
 
+import java.util.Iterator;
 import java.util.LinkedHashMap;
 import java.util.Map;
 
+import org.apache.commons.lang3.StringUtils;
+
+import com.google.common.base.Splitter;
+import com.sequenceiq.cloudbreak.logger.PartialMessageMetadata;
+
 import ch.qos.logback.classic.spi.ILoggingEvent;
 import ch.qos.logback.contrib.json.classic.JsonLayout;
 import ch.qos.logback.core.CoreConstants;
 
 public class CustomJsonLayout extends JsonLayout {
 
+    private static final String PARTIAL_MESSAGE_FIELD = "partial_message";
+
+    private static final String PARTIAL_CHUNK_ID_FIELD = "partial_id";
+
+    private static final String PARTIAL_CHUNK_INDEX_FIELD = "partial_ordinal";
+
+    private static final String PARTIAL_LAST_FIELD = "partial_last";
+
+    private static final String PARTIAL_MESSAGE_FLAG = "true";
+
     private final String contextName;
 
-    CustomJsonLayout(String contextName) {
+    private final Integer maxChunkLength;
+
+    CustomJsonLayout(String contextName, Integer maxChunkLength) {
         this.contextName = contextName;
+        this.maxChunkLength = maxChunkLength;
     }
 
     public String doLayout(ILoggingEvent event, String fullLogMessage, String loggerNameFilter) {
-        Map map = toJsonMap(event, fullLogMessage, loggerNameFilter);
-        if (map == null || map.isEmpty()) {
+        if (StringUtils.length(fullLogMessage) > this.maxChunkLength) {
+            StringBuilder stringBuilder = new StringBuilder();
+            String chunkId = Integer.toHexString(event.hashCode());
+            Iterable<String> messages = Splitter.fixedLength(this.maxChunkLength).split(fullLogMessage);
+            int currentIndex = 0;
+            Iterator<String> messageIterator = messages.iterator();
+            while (messageIterator.hasNext()) {
```

Comment:
you can save one line here with:
`for (int currentIndex = 0; messageIterator.hasNext(); currentIndex++) {`